chainstate/test-framework/src/utils.rs

```diff
@@ -264,3 +265,42 @@ fn new_token_burn_output(
         OutputPurpose::Burn,
     )
 }
+
+// TODO: Replace by a proper UTXO set abstraction
+// (https://github.com/mintlayer/mintlayer-core/issues/312).
+pub type TestBlockUtxos = Vec<(OutPointSourceId, Vec<TxOutput>)>;
```

Comment:
I would make this a map to protect from accidental duplication.

Reply:
OK, updated. Though loading outputs looks a bit ugly in the tests now.

Reply:
The following code got me thinking. 
```rust 
let issuance_outpoint_id =
             tf.outputs_from_genblock(block.get_id().into()).keys().next().unwrap().clone();
```
First I thought that it was not a good idea to switch to a map. Many tests rely on the ordering of txs. And I was really surprised that no tests failed because of this. And it just looked ugly anyway. 

But then the more I looked at that TestBlockInfo the more ugly it got. For example:
```rust
        let issuance_outpoint_id = TestBlockInfo::from_block(&block).txns[0].0.clone(); 
```
It's insane: collecting all the outputs from all the txs just to get id for the first tx! Instead of something like `block.transaction().get(0).get_id()`.

I think that in most cases `TestBlockInfo` is heavily misused (as well as `BlockOutputs` because it doesn't solve the root cause). Imo what we actually need is a  way to **get transaction id using block id and index**. This is handy because when constructing blocks with BlockBuilder + TransactionBuilder you won't get ids of txs to be able to use them later in the test.

I suggest adding dedicated method to test framework: `get_transaction(id: Id<Block>, index)`. With it I believe that 90% of the smelly code goes away.

`BlockOutputs` still can be used for other 10% of cases that actually use outputs.

P.S. there is almost exact copy of `TestBlockInfo` in `p2p` [dizzy]

Reply:
> P.S. there is almost exact copy of TestBlockInfo in p2p [dizzy]

Test client still isn't used in `p2p` because it (or the `p2p` code) requires some adaptation. We have an issue for that: https://github.com/mintlayer/mintlayer-core/issues/182

Reply:
Answering you, @azarovh , this is what happens when someone created a very bad design and it trickles down everywhere. Hopefully we'll fix this. I think you're mostly right, but we want to give this some good thought. 

We probably do need a way to get all utxos. But if most of the calls just need a single utxo, simplifying this is a good call.